### PR TITLE
PPU LLVM: properly compile patched instructions (need testing)

### PR DESCRIPTION
**Pull request by Nekotekina**
Use some information exported from the patch engine.

Also possibly fix SPRX patching.

### CODE REVIEW

File: Utilities/bin_patch.cpp [hunk 1]
```diff
@@ -505,26 +505,27 @@ void patch_engine::append_title_patches(const std::string& title_id)
...
 
-usz patch_engine::apply_with_ls_check(const std::string& name, u8* dst, u32 filesz, u32 ls_addr)
+std::basic_string<u32> patch_engine::apply_with_ls_check(const std::string& name, u8* dst, u32 filesz, u32 ls_addr)
 {
 	return apply_patch<true>(name, dst, filesz, ls_addr);
 }
 
-template <bool check_local_storage>
-static usz apply_modification(const patch_engine::patch_info& patch, u8* dst, u32 filesz, u32 ls_addr)
+template <bool CheckLS>
```
[the remark this answers is not in the document]
Comment (Nekotekina) [reply Utilities/bin_patch.cpp:518]: I don't like long names, idk what to do.

**Comment by Nekotekina** [comment]
Needs more debugging...

**Comment by Oschowa** [comment]
Latest version of this PR seems to work fine with patched Demon's Souls. Also tested (patched) Drag-on Dragoon 3 and Atelier Ayesha and those were fine as well.

**Comment by Nekotekina** [comment]
Thanks for testing, maybe it's ready for merge.

**Comment by RainbowCookie32** [comment]
Seems to also fix #9727, boot times are back to normal.

**Pull request merged**